tools/indextemplate/indextemplate.go

```diff
@@ -133,14 +133,12 @@ func (engine *Engine) resolveURI(parsedName map[string]string) (uri *url.URL, er
 	return uri, nil
 }
 
-func (engine *Engine) handleIndex(mediatype string, response *http.Response, parsedName map[string]string) (roots []refengine.MerkleRoot, err error) {
+func (engine *Engine) handleIndex(response *http.Response, parsedName map[string]string) (roots []refengine.MerkleRoot, err error) {
 	roots = []refengine.MerkleRoot{}
 	var index v1.Index
 
-	// FIXME: check response content type (and charset?)
-
 	if err := json.NewDecoder(response.Body).Decode(&index); err != nil {
-		logrus.Errorf("%s claimed to return %s, but the response schema did not match: %s", response.Request.URL, mediatype, err)
+		logrus.Errorf("%s claimed to return, but the response schema did not match: %s", response.Request.URL, err)
```

Comment:
I think your new error message is less helpful.  Maybe we can move all the `mediatype` handling out of `Get` over to where you're removing the FIXME.  Then you don't need to pass `mediatype` to `handleIndex`, but we still get the claimed media type in this error.  Also, `handleIndex` is unit tested, while the full `Get` is not.